src/System.Private.CoreLib/shared/System/IO/StreamWriter.cs

```diff
@@ -476,6 +476,124 @@ public override void WriteLine(ReadOnlySpan<char> value)
             }
         }
 
+        private void WriteFormatHelper(string format, ParamsArray args, bool appendNewLine)
+        {
+            StringBuilder sb =
+                StringBuilderCache.Acquire(format.Length + args.Length * 8)
+                .AppendFormatHelper(null, format, args);
+
+            StringBuilder.ChunkEnumerator chunks = sb.GetChunks();
+
+            bool more = chunks.MoveNext();
+            while (more)
+            {
+                ReadOnlySpan<char> current = chunks.Current.Span;
+                more = chunks.MoveNext();
+
+                // If final chunk, include the newline if needed
+                WriteSpan(current, appendNewLine: more ? false : appendNewLine);
+            }
+
+            StringBuilderCache.Release(sb);
+        }
+
+        public override void Write(string format, object arg0)
+        {
+            if (GetType() == typeof(StreamWriter))
+            {
+                WriteFormatHelper(format, new ParamsArray(arg0), appendNewLine: false);
+            }
+            else
+            {
+                base.WriteLine(format, arg0);
```

Comment:
WriteLine -> Write

Reply:
If this is not caught by tests, we may want to add a functional tests to cover this path.

Reply:
Whoops- I'll make sure there is a test.